logstash-core-event-java/src/main/java/com/logstash/Event.java

```diff
@@ -139,8 +145,15 @@ public void setField(String reference, Object value) {
         } else if (reference.startsWith(METADATA_BRACKETS)) {
             this.metadata_accessors.set(reference.substring(METADATA_BRACKETS.length()), value);
         } else {
-            this.accessors.set(reference, value);
+            this.accessors.set(reference, setableValue(value));
+        }
+    }
+
+    private Object setableValue(Object value) {
```

Comment:
I am not sure I like `setableValue`, what about simply `rubyJavaValue` ? I know it can also return Map and List... not sure ... ?